tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/ConsumerGroupHelper.cs

```diff
@@ -0,0 +1,27 @@
+﻿// <copyright file="ConsumerGroupHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka;
+
+internal class ConsumerGroupHelper
+{
+    // A map between Kafka Consumer<TKey,TValue> and the corresponding consumer group
+    private static readonly ConditionalWeakTable<object, string> ConsumerToGroupIdMap = new();
+
+    public static void SetConsumerGroup(object consumer, string groupId)
+    {
+        ConsumerToGroupIdMap.GetValue(consumer, x => groupId);
```

Comment:
why not `ConsumerToGroupIdMap.AddOrUpdate(consumer, groupId);` to avoid the lambda?

Reply:
`AddOrUpdate` doesn't exist < `netcoreapp3.1`. Will add an `#if`, thanks 🙂